packages/adapter-surrealdb/src/index.ts

```diff
@@ -14,138 +14,197 @@
+  emailVerified: user.emailVerified ?? undefined,
+})
 
 /** @internal */
 // Convert AdapterAccount to DB object
-const accountToDoc = (account: AdapterAccount): Omit<AccountDoc, "id"> => {
-  const doc = {
-    ...account,
-    userId: `user:${toSurrealId(account.userId)}`,
-  }
-  return doc
-}
+const accountToDoc = (account: AdapterAccount): Omit<AccountDoc, "id"> => ({
+  ...account,
+  userId: new RecordId("user", account.userId.replace("user:", "")),
+})
+
+/** @internal */
+// Convert AdapterAuthenticator to DB object
+const authenticatorToDoc = (
+  authenticator: AdapterAuthenticator
+): Partial<AuthenticatorDoc> => ({
+  ...authenticator,
+  userId: new RecordId("user", authenticator.userId.replace("user:", "")),
+})
 
 /** @internal */
 // Convert AdapterSession to DB object
 export const sessionToDoc = (
   session: AdapterSession
-): Omit<SessionDoc, "id"> => {
-  const doc = {
-    ...session,
-    expires: session.expires.toISOString(),
-  }
-  return doc
-}
+): Partial<SessionDoc<RecordId<string>>> => ({
+  ...session,
+  userId: new RecordId("user", session.userId.replace("user:", "")),
+  expires: session.expires,
+})
 
-export const toSurrealId = (id: string) => {
-  if (/^⟨.+⟩$/.test(id)) {
-    return id
-  } else {
-    return `⟨${id}⟩`
-  }
-}
+/** @internal */
+// Convert VerificationToken to DB object
+const verificationTokenToDoc = (
+  account: VerificationToken
+): Omit<VerificationTokenDoc, "id"> => ({
+  ...account,
+})
 
-export const toId = (surrealId: string) => {
-  return surrealId.replace(/^⟨(.+)⟩$/, "$1")
+/** @internal */
+/**
+ * Removes all undefined fields in an object
+ * @param obj
+ * @returns
+ */
+function removeUndefinedFields<T>(obj: T): Partial<T | null> {
+  if (typeof obj !== "object" || obj === null) {
+    return obj
+  }
+  for (const key in obj) {
+    if (obj[key] === undefined) {
+      delete obj[key]
+    } else if (typeof obj[key] === "object" && obj[key] !== null) {
+      removeUndefinedFields(obj[key])
+    }
+  }
+  return obj
 }
 
-export function SurrealDBAdapter<T>(
-  client: Promise<Surreal | ExperimentalSurrealHTTP<T>>
+export function SurrealDBAdapter(
+  client: Promise<Surreal>
   // options = {}
 ): Adapter {
   return {
-    async createUser(user: Omit<AdapterUser, "id">) {
-      const surreal = await client
-      const doc = userToDoc(user)
-      const userDoc = await surreal.create<UserDoc, Omit<UserDoc, "id">>(
-        "user",
-        doc
-      )
-      if (userDoc.length) {
-        return docToUser(userDoc[0])
-      }
+    async createUser(user: Partial<AdapterUser>) {
+      try {
+        const surreal = await client
+        const doc = userToDoc(user)
+        const userDoc = await surreal.create<UserDoc, Omit<UserDoc, "id">>(
+          "user",
+          doc
+        )
+        if (userDoc.length) {
+          return docToUser(userDoc[0])
+        }
+      } catch {}
       throw new Error("User not created")
```

Comment:
@dvanmali Finally getting around to try out your changes and noticed something and had a suggestion. this error always gets thrown. If you have time it would also be great if you could use the logger to log the errors being caught as well.

Reply:
Will do, I should have some time tonight or tomorrow. Thanks!

Reply:
I missed the return just above, so it turns out the error doesn't always get thrown. I looked into the logger, and I guess it's not easily accessible. Some better logging would be helpful though, but perhaps that's more on AuthJS side as a stacktrace doesn't even get logged.